src/getMessageAuthor.ts

```diff
@@ -88,3 +119,40 @@ function authorFromGitHubContext(): MessageAuthor | null {
     icon_url: sender.avatar_url
   }
 }
+
+async function getSenderFromPRMerger(
+  octokit: OctokitClient
+): Promise<MessageAuthor> {
+  const payload = context.payload
+
+  const matches = (payload.head_commit?.message as string).match(/\(#(\d+)\)$/)
+
+  if (!matches) {
+    throw new Error(
+      `Failed to parse PR number from commit message: '${payload.head_commit?.message}'.`
+    )
+  }
+
+  const prNumber = Number(matches[1])
+
+  if (!payload.repository) {
+    throw new Error('WebhookPayload does not include repository information')
+  }
+
+  const mergedBy = (
+    await octokit.rest.pulls.get({
```

Comment:
FYI, we have a [ticket](https://app.shortcut.com/fieldguide/story/52673/document-mininum-required-perms-to-run-action-slack-deploy-pipeline) to document required GitHub permissions for this action (which we can probably sort out in a followup, unless you're inspired to figure that out now).